### PR TITLE
Track gaps in playtime in saved vars

### PR DESCRIPTION
**Pull request by aaronma37**
Tracks gaps in playtime between the addon and the server's requestPlayTime().  Offsets against accumulated differences between played_time and time_tracked since this differs by about 1-3 seconds every time the player logs in

* Records violation if the gap is longer than 30 seconds
* On average, it looked like gaps were between 1-3 seconds (maybe from loading)
* Warning message example below (generated by changing threshold to 0)

image

### CODE REVIEW

File: Hardcore.lua [hunk 1]
```diff
@@ -27,10 +27,13 @@ Hardcore_Settings = {
 
 --[[ Character saved variables ]]--
 Hardcore_Character = {
-	time_tracked = 0,
-	time_played = 0,
+	time_tracked = 0, 		-- seconds
+	time_played = 0, 		-- seconds
+	accumulated_time_diff = 0, 	-- seconds
+	tracked_played_perc = nil,
```
Comment (blenderer) [inline Hardcore.lua:33]: Nitpick: Use full word in variable. Not sure what "perc" means immediately looking at this

File: Hardcore.lua [hunk 2]
```diff
@@ -65,6 +68,10 @@ local Last_Attack_Source = nil
 local PICTURE_DELAY = .65
 local HIDE_RTP_CHAT_MSG = false
 local STARTED_BUBBLE_HEARTH_INFO = nil
+local RECV_FIRST_PLAYED_TIME = false
```
Comment (blenderer) [inline Hardcore.lua:71]: Nitpick: Use full word in variable

RECV = Received?

**Review by RobertB4 — APPROVED** [review]
Looks good - nits mentioned by blenderer would be nice if you could fix them, I'm a fan of readable variables names too :)

**Comment by aaronma37** [comment]
> Looks good - nits mentioned by blenderer would be nice if you could fix them, I'm a fan of readable variables names too :)

Done!

**Review by RobertB4 — APPROVED** [review]
👍

**Review by blenderer — APPROVED** [review]

**Pull request merged**